shotover-proxy/tests/cassandra_int_tests/mod.rs

```diff
@@ -541,67 +589,51 @@ async fn test_cassandra_request_throttling() {
 
     // this batch set should be allowed through
     {
-        let mut batch = Batch::new(BatchType::LOGGED);
+        let mut queries: Vec<(String, i32)> = vec![];
         for i in 0..25 {
-            let statement = format!("INSERT INTO test_keyspace.my_table (id, lastname, firstname) VALUES ({}, 'text', 'text')", i);
-            batch.add_statement(&stmt!(statement.as_str())).unwrap();
+            queries.push(("INSERT INTO test_keyspace.my_table (id, lastname, firstname) VALUES (?, 'text', 'text')".into(), i));
         }
-        connection.execute_batch(&batch);
+        connection.execute_batch(queries);
     }
 
     std::thread::sleep(std::time::Duration::from_secs(1)); // sleep to reset the window
 
     // this batch set should not be allowed through
     {
-        let mut batch = Batch::new(BatchType::LOGGED);
+        let mut queries: Vec<(String, i32)> = vec![];
         for i in 0..60 {
-            let statement = format!("INSERT INTO test_keyspace.my_table (id, lastname, firstname) VALUES ({}, 'text', 'text')", i);
-            batch.add_statement(&stmt!(statement.as_str())).unwrap();
+            queries.push(("INSERT INTO test_keyspace.my_table (id, lastname, firstname) VALUES (?, 'text', 'text')".into(), i));
         }
-        let result = connection.execute_batch_expect_err(&batch);
-        assert!(matches!(
-            result,
-            Error(
-                ErrorKind::CassErrorResult(cassandra_cpp::CassErrorCode::SERVER_OVERLOADED, ..),
```

Comment:
I dont want to regress on important testing here.
The error code is the actual important bit here that will alter how the client processes the error so we need to at least be testing that.
Maybe execute_batch_expect_err should return a:
```rust
struct CassandraError {
    code: i32,
    message: String 
}
```
That we map cdrs-tokio and casandra_cpp types into.

The `code` field could be given a custom enum instead of an i32 but not necessary.